File: src/main/java/net/ndrei/teslapoweredthingies/machines/fluidburner/FluidBurnerCoolantRecipe.java
package net.ndrei.teslapoweredthingies.machines.fluidburner;

import net.minecraftforge.fluids.Fluid;

/**
 * Created by CF on 2017-01-09.
 */
public class FluidBurnerCoolantRecipe {
    public final Fluid fluid;
    public final int amount;
    public final float timeMultiplier;

    public FluidBurnerCoolantRecipe(Fluid fluid, int amount, float timeMultiplier) {
        this.fluid = fluid;
        this.amount = amount;
        this.timeMultiplier = timeMultiplier;
    }
}

File: src/main/kotlin/net/ndrei/teslapoweredthingies/machines/fluidburner/FluidBurnerCoolantRecipe.kt
package net.ndrei.teslapoweredthingies.machines.fluidburner

import net.minecraft.util.ResourceLocation
import net.minecraftforge.fluids.Fluid
import net.ndrei.teslapoweredthingies.MOD_ID
import net.ndrei.teslapoweredthingies.common.BaseTeslaRegistryEntry

/**
 * Created by CF on 2017-06-30.
 */
class FluidBurnerCoolantRecipe(val fluid: Fluid, val amount: Int, val timeMultiplier: Float)
    : BaseTeslaRegistryEntry<FluidBurnerCoolantRecipe>(FluidBurnerCoolantRecipe::class.java, ResourceLocation(MOD_ID, "fluid_${fluid.name}"))
